### PR TITLE
modules/SceLibKernel: implement SceIoRename.

### PR DESCRIPTION
**Pull request by Zangetsu38**
# About
- Implment SceIoRename
Tested in both case, rename in same folder, and rename to other folder, is fine.

### CODE REVIEW

File: vita3k/io/src/io.cpp [hunk 1]
```diff
@@ -619,6 +619,47 @@ int remove_file(IOState &io, const char *file, const std::wstring &pref_path, co
...
+        LOG_ERROR("File does not exist at path: {} (target path: {})", emulated_old_path.string(), old_name);
+        return IO_ERROR(SCE_ERROR_ERRNO_ENOENT);
+    }
+
+    const auto emulated_new_path = device::construct_emulated_path(device, translated_new_path, pref_path, io.redirect_stdio);
+
+    LOG_TRACE_IF(log_file_op, "{}: Renaming file {} to {} ({} to {})", export_name, old_name, new_name, emulated_old_path.string(), emulated_new_path.string());
+
+    boost::system::error_code error_code{};
+    fs::detail::rename(emulated_old_path, emulated_new_path, &error_code);
```
Comment (Macdu) [inline vita3k/io/src/io.cpp:652]:
```suggestion
    fs::rename(emulated_old_path, emulated_new_path, &error_code);
```

fs::detail is not portable (boost only)

Comment (Macdu) [reply vita3k/io/src/io.cpp:652]: Other part of the code using it will have to be changed.

Comment (Zangetsu38) [reply vita3k/io/src/io.cpp:652]: ah i see hmm but i no can check error code so ? 🤔

Comment (Macdu) [reply vita3k/io/src/io.cpp:652]: image
You should be able to

Comment (Zangetsu38) [reply vita3k/io/src/io.cpp:652]: ah yes.. i hgave no understund, it said me no can, i remove, and after i re set and said is ok, visual studio drink

Comment (Zangetsu38) [reply vita3k/io/src/io.cpp:652]: ah no çok, it no want i using &error_code but pass with just error_code

Comment (Zangetsu38) [reply vita3k/io/src/io.cpp]: ok, so done

**Review by Macdu — APPROVED** [review]
Make sure for the CI to be done before merging, just in case.

**Comment by Zangetsu38** [comment]
oki 👌

**Pull request merged**